community/bolt/src/test/java/org/neo4j/bolt/v1/runtime/internal/ErrorReporterTest.java

```diff
@@ -56,4 +57,50 @@ public void shouldReportUnknownErrors()
         provider.assertExactly(
                 inLog( "userlog" ).error( CoreMatchers.containsString( error.reference().toString() ) ) );
     }
+
+    @Test
+    public void shouldNotReportOOMErrors()
+    {
+        // Given
+        AssertableLogProvider provider = new AssertableLogProvider();
+        ErrorReporter reporter =
+                new ErrorReporter( provider.getLog( "userlog" ), new UsageData( mock( JobScheduler.class ) ) );
+
+        Throwable cause = new OutOfMemoryError( "memory is fading" );
+        Neo4jError error = Neo4jError.from( cause );
+
+        // When
+        reporter.report( error );
+
+        // Then
+        assertThat( error.status(), equalTo( (Status) Status.General.OutOfMemoryError ) );
+        assertThat( error.message(), equalTo(   "There is not enough memory to perform the current task. Please try increasing " +
+                                                             "'dbms.memory.heap.max_size' in 'conf/neo4j-wrapper.conf' or if you are running an embedded " +
+                                                             "installation increase the heap by using '-Xmx' command line flag." ));
+        provider.assertNoLoggingOccurred();
+    }
+
+    @Test
+    public void shouldNotReportStackOverflowErrors()
+    {
+        // Given
+        AssertableLogProvider provider = new AssertableLogProvider();
+        ErrorReporter reporter =
+                new ErrorReporter( provider.getLog( "userlog" ), new UsageData( mock( JobScheduler.class ) ) );
+
+        Throwable cause = new StackOverflowError( "some rewriter is probably not tail recursive" );
+        Neo4jError error = Neo4jError.from( cause );
+
+        // When
+        reporter.report( error );
+
+        // Then
+        assertThat( error.status(), equalTo( (Status) Status.General.OutOfMemoryError ) );
+        assertThat( error.message(), equalTo(  "There is not enough stack size to perform the current task. This is generally considered to be a " +
+                                                            "database error, so please contact Neo4j support. You could try increasing the stack size: " +
+                                                            "for example to set the stack size to 2M, add `dbms.jvm.additional=-Xss2M' to " +
+                                                            "'conf/neo4j-wrapper.conf' or if you are running an embedded installation just add -Xss2M as " +
+                                                            "command line flag." ));
+        provider.assertNoLoggingOccurred();
```

Comment:
I think that is right though - we should force both places to be changed manually, that guards against accidentally changing the message (eg. if we throw a similar exception somewhere else). This tests that the surface behavior is exactly as intended, and since it's in a test it won't go out of sync, it just forces an additional manual review whenever you make a change that bubbles up to the end user.